patches/server/0067-lithium-skip-ticking-block-entities-that-are-doing-n.patch

```diff
@@ -0,0 +1,1348 @@
++            ((BlockEntitySleepTracker) this.world).setAwake(this, false);
++        }
++        // Yatopia end
+     }
+ 
+     public void doOpenLogic() {
+@@ -197,6 +226,7 @@ public class TileEntityChest extends TileEntityLootable { // Paper - Remove ITic
+ 
+     @Override
+     public boolean setProperty(int i, int j) {
++        this.checkWakeUp(); // Yatopia - lithium: skip ticking block entities that are doing nothing
+         if (i == 1) {
+             this.viewingCount = j;
+             return true;
+@@ -228,7 +258,7 @@ public class TileEntityChest extends TileEntityLootable { // Paper - Remove ITic
+             // CraftBukkit end
+             this.onOpen();
+         }
+-
++        this.checkWakeUp(); // Yatopia - lithium: skip ticking block entities that are doing nothing
+     }
+ 
+     @Override
+@@ -249,6 +279,7 @@ public class TileEntityChest extends TileEntityLootable { // Paper - Remove ITic
+             // CraftBukkit end
+             this.onOpen();
+         }
++        this.checkWakeUp(); // Yatopia - lithium: skip ticking block entities that are doing nothing
+ 
+     }
+ 
+diff --git a/src/main/java/net/minecraft/server/TileEntityEnchantTable.java b/src/main/java/net/minecraft/server/TileEntityEnchantTable.java
+index e3de65b58a599b375b3be7470d918038b5379471..ad59d7d4496ce2eaf7f2adbec14367f2c6f4d7a6 100644
+--- a/src/main/java/net/minecraft/server/TileEntityEnchantTable.java
++++ b/src/main/java/net/minecraft/server/TileEntityEnchantTable.java
+@@ -2,6 +2,7 @@ package net.minecraft.server;
+ 
+ import java.util.Random;
+ import javax.annotation.Nullable;
++ import me.jellysquid.mods.lithium.common.world.blockentity.BlockEntitySleepTracker;
+ 
+ public class TileEntityEnchantTable extends TileEntity implements INamableTileEntity, ITickable {
+ 
+@@ -44,6 +45,7 @@ public class TileEntityEnchantTable extends TileEntity implements INamableTileEn
+     @Override
+     public void tick() {
+         if (!org.yatopiamc.yatopia.server.YatopiaConfig.shouldTickEnchantingTables) { return; } // Yatopia - TE optimizations
++        ((BlockEntitySleepTracker) this.world).setAwake(this, false); // Yatopia
+         this.j = this.i;
+         this.l = this.k;
+         EntityHuman entityhuman = this.world.a((double) this.position.getX() + 0.5D, (double) this.position.getY() + 0.5D, (double) this.position.getZ() + 0.5D, 3.0D, false);
+diff --git a/src/main/java/net/minecraft/server/TileEntityEnderChest.java b/src/main/java/net/minecraft/server/TileEntityEnderChest.java
+index 80440aa24f546dd151e89b705322d9c3e906b139..94c2bfce5059d784655d729f384c70b165432311 100644
+--- a/src/main/java/net/minecraft/server/TileEntityEnderChest.java
++++ b/src/main/java/net/minecraft/server/TileEntityEnderChest.java
+@@ -1,17 +1,42 @@
+ package net.minecraft.server;
++import me.jellysquid.mods.lithium.common.world.blockentity.BlockEntitySleepTracker;
+ 
+ public class TileEntityEnderChest extends TileEntity { // Paper - Remove ITickable
```

Comment:
Is it necessary to do extra logic if ender chest isnt tickable?

Reply:
Good catch.